ansible_base/lib/dynamic_config/__init__.py

```diff
@@ -0,0 +1,257 @@
+    """Export the settings to the module that called this function.
+
+    module_name is usually `__name__` and is used to get the module object
+    arbitrary module objects can be passed as well so testing can be done on
+    a fake module object.
+
+    In a djangoapp/settings.py this will take all variables from the Dynaconf
+    instance and set them as attributes in the settings module
+    so they can be accessed as `settings.VARIABLE_NAME` from django.conf.settings.
+    """
+    if isinstance(module_name, str):
+        object_to_populate = sys.modules[module_name]
+    else:
+        object_to_populate = module_name
+
+    settings.populate_obj(
+        object_to_populate,
+        internal=False,
+        ignore=["IS_DEVELOPMENT_MODE"],
+        convert_to_dict=True,
+    )
+
+
+def load_standard_settings_files(settings: Dynaconf):
+    """Load the standard settings files for Ansible Automation Platform.
+
+    This function loops the `ANSIBLE_STANDARD_SETTINGS_FILES` list in the settings
+    and loads each file into the settings object without considering the current env,
+    as these files are meant to be environment agnostic.
+
+    NOTE: this function must be replaced by future implementation
+    of Dynaconf's `settings.load_file` that will support envless loading.
+    """
+    loader = BaseLoader(
+        obj=settings,
+        env=settings.current_env,
+        identifier="standard_settings_loader",
+        extensions=YAML_EXTENSIONS,
+        file_reader=yaml.safe_load,
+        string_reader=yaml.safe_load,
+        validate=False,
+    )
+    for path in settings.ANSIBLE_STANDARD_SETTINGS_FILES:
+        if not Path(path).exists():
+            continue
+        data = loader.get_source_data([str(path)])
+        loader._envless_load(data)
+
+
+def load_envvars(settings: Dynaconf):
+    """Loads environment variables into the settings object.
+
+    This function exists to be called on-demand after all settings are loaded
+    """
+    env_loader.load(settings, identifier="dab.dynamic_config")
+
+
+def _check_options(options):
+    """Raise Error if invalid options are passed.
+
+    Invalid options are those that are not allowed in the factory function,
+    and must be defined here by Django Ansible Base.
+    """
+    invalid_options = {
+        "envvar_prefix",  # defined based on app name
+        "env_switcher",  # defined based on app name
+        "root_path",  # defined dynamically by factory
+        "validators",  # Cannot be defined because would trigger eager validation
+    }
+    if invalid_options & set(options.keys()):
+        raise ImproperlyConfigured(f"Invalid Dynaconf options: {invalid_options}")
+
+
+def validate(settings: Dynaconf):
+    """Validate the settings according to the validators registered.
+    This function operates on a clone of the settings object to avoid
+    validation to write to the inspection data.
+    """
+    settings.dynaconf_clone().validators.validate()
```

Comment:
Do we really want to avoid writing? I think the Validator's cast option is very useful to convert a value rather than rejecting it.